### DISCUSSION TITLE
ESP32 hardware connections

### DISCUSSION THREAD

**Discussion by suysh** [emsesp/EMS-ESP32#1039] [General]
In the documentation I found the connections for ESP8266 Wemos D1 mini.
![image](https://user-images.githubusercontent.com/43004004/108957709-3d327b80-7672-11eb-83ac-c859f31f1000.png)
To upgrade to ESP32 I can't find the needed connections. I have this board:
![image](https://user-images.githubusercontent.com/43004004/108957843-74089180-7672-11eb-86fa-26d72f5dd1b8.png)
![image](https://user-images.githubusercontent.com/43004004/108957873-808cea00-7672-11eb-8c2b-5b9834115c7f.png)
Kind regards, Hans

**Comment by MichaelDvP** [comment]
This board is pin compatible to Wemos D1 Mini if you use the white framed pins and parts side up. 
The settings are default for esp32: 
TX: io5, RX: io23, LED: io2 (inverted), DALLAS: io18

**Comment by proddy** [comment]
Also good to add to https://github.com/proddy/EMS-ESP/issues/690

**Comment by bbqkees** [comment]
I just updated my wiki:
https://bbqkees-electronics.nl/wiki/gateway/upgrade-the-gateway-to-an-esp32.html

**Comment by glitter-ball** [comment]
@bbqkees - thanks! Brilliant docs as ever and I’ve just ordered an ESP32 to give it a go.

If, in time, someone finds a case which fits (or could provide 3D print file) that would be perfect but really happy to give it a go like this.

**Comment by bbqkees** [comment]
@glitter-ball If you cut the case lid as described it actually fits together pretty nicely afterwards.
The only thing that will be missing is the side piece which is not really that important I think, especially if you use the USB port.
If you really want you can also cut the side piece in half so it will fit again as well. Then only the ESP32 board will stick out a little bit.

**Comment by suysh** [comment]
I upgraded today to ESP32 and it works as now described in the excellent documentation.
